ml/algebra/matrix_ops.h

```diff
@@ -155,6 +155,41 @@ diag(const distribution<Float> & d)
 }
 
 template<typename Float>
+boost::multi_array<Float, 2>
+outerProd(const distribution<Float> & d)
```

Comment:
Ugh, do we really need to explicitly instantuate an outer product?  Normally I prefer to factor it out so that we don't end up creating it.